src/main/java/com/ericsson/ei/controller/QueryMissedNotificationControllerImpl.java

```diff
@@ -25,34 +24,32 @@
 import org.springframework.web.bind.annotation.CrossOrigin;
 import org.springframework.web.bind.annotation.RequestParam;
 
-import com.ericsson.ei.controller.model.QueryResponse;
-import com.ericsson.ei.queryservice.ProcessMissedNotification;
+import java.util.List;
 
 /**
  * This class represents the REST GET mechanism to extract the aggregated data
  * on the basis of the SubscriptionName from the Missed Notification Object.
- * 
  */
 @Component
 @CrossOrigin
-public class MissedNotificationControllerImpl implements MissedNotificationController {
+public class QueryMissedNotificationControllerImpl implements QueryMissedNotificationController {
 
-    static Logger log = (Logger) LoggerFactory.getLogger(MissedNotificationControllerImpl.class);
+    private final static Logger LOGGER = (Logger) LoggerFactory.getLogger(QueryMissedNotificationControllerImpl.class);
 
     @Autowired
     private ProcessMissedNotification processMissedNotification;
 
     /**
      * This method is responsible for the REST GET mechanism to extract the data on
      * the basis of the SubscriptionName from the Missed Notification Object.
-     * 
+     *
      * @param subscriptionName
      * @return ResponseEntity
      */
     public ResponseEntity<QueryResponse> getQueryMissedNotifications(
```

Comment:
I found our rest services should return ResponseEntity<QueryResponse> but we are returning to ResponseEntity(String, HttpStatus.OK);
For now please change in current modified code to ResponseEntity<QueryResponse>, later code refactor we can do all rest endpoints

Reply:
Could you please clarify how it should looks like? because i don't clearly understand

Reply:
QueryResponse queryResponse = new QueryResponse();
queryResponse .setResponseEntity(response.toString())
return new ResponseEntity(queryResponse , HttpStatus.OK);